### PR TITLE
Migrate travis to ADO

### PR DESCRIPTION
**Pull request by ameyapat**
## Proposed changes
Migrating travis pipeline to Azure Dev Ops. 
Pipeline located at \CI\Apple\PR Validations in Azure Dev Ops.

Transferred step to produce docs from travis.yml to release pipeline on master since, docs were being generated only if branch was master/main.

The new pipeline will spawn 2 jobs in parallel. One to build/test iOS related targets & other for macOS targets.
## Type of change

- [ ] Feature work
- [ ] Bug fix
- [x] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: azure_pipelines/pr-validation.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Pipeline will be triggered for PR & any updates on the PR on all branches
```
Comment (kaisong1990) [inline azure_pipelines/pr-validation.yml:1]: This pull request does not update CHANGELOG.md.

Please consider if this change would be noticeable to a partner or user and either update CHANGELOG.md or resolve this conversation.

File: azure_pipelines/pr-validation.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+# Define parallel jobs that run build script for specified targets
+jobs:
+- job: 'Validate_Pull_Request'
+  strategy:
+    maxParallel: 2
+    matrix:
+      IOS_FRAMEWORK: 
+        target: "iosFramework iosTestApp sampleIosApp sampleIosAppSwift"
+#      IOS_TEST_APP:
```
Comment (oldalton) [inline azure_pipelines/pr-validation.yml:20]: Remove commented out code

File: build_docs.sh [hunk 3]
```diff
@@ -1,14 +1,11 @@
 #!/bin/bash
+gem install jazzy
+echo -e "Copying MSAL public files"
```
Comment (oldalton) [inline build_docs.sh:3]: Should this be done only for master branch? I don't think we want to regenerate docs on each PR.

**Review by oldalton — APPROVED** [review]

File: azure_pipelines/pr-validation.yml (hunk 2 above)
Comment (ameyapat) [reply azure_pipelines/pr-validation.yml]: Removed

File: build_docs.sh (hunk 3 above)
Comment (ameyapat) [reply build_docs.sh:3]: Yes, it is only done for master since it will be executed in spm-framework.yml, which is our final release pipeline.

**Pull request merged**